doc/design/design_model_zoo.md

```diff
@@ -0,0 +1,99 @@
+   ```sql
+   SELECT * FROM iris.transfer_iris_samples
+   TRAIN sqlflow.org/modelzoo/iris_dnn_128x32
+   USING sqlflow.org/modelzoo/iris_dnn_128x32
+   INTO modeldb.my_iris_dnn_model;
+   ```
+1. Fine-tune the pre-trained model:
+   ```sql
+   SELECT * FROM iris.train
+   TRAIN sqlflow.org/modelzoo/iris_dnn_128x32
+   WITH model.learning_rate=0.001, model.learning_rate_decay="cosine_decay" ...
+   INTO modeldb.my_iris_dnn_model_fine_tune;
+   ```
+
+## The Model Zoo Hosting Service
+
+The model zoo hosting service is a file service that can be accessed from the internet.
+It serves all available models and corresponding pre-trained weight. We may need to use
+a [CDN](https://en.wikipedia.org/wiki/Content_delivery_network) service if the traffic
+becomes large.
+
+Each model is saved on the server under a unique directory like: `sqlflow.org/modelzoo/iris_dnn_128x32`.
+The directory name is responsible to explain the model's type, network structure and which dataset is
+used to train the pre-trained weights. You can access `sqlflow.org/modelzoo/iris_dnn_128x32/README.md`
+from the browser to get the model's full documentation. All models under `sqlflow.org/modelzoo` are
+developed under `https://github.com/sql-machine-learning/models` weights is only stored under
+`sqlflow.org` but not under github.
+
+The content of the directory should be like:
+
+```
+iris_dnn_128x32
+   - README.md     # model documents.
+   - model_def.py  # python file of model definition (Tensorflow Estimator model or Keras model).
+   - model/        # the saved Tensorflow/Keras/XGBoost model.
+   - columns       # saved codegen.FeatureColumn meta for parsing and extracting features.
+   - some_deps.py  # if model_def.py have dependent python source files just put them in the same folder.
+```
+
+- Note that we save the `columns` file so that we can check if user-provided data is of the same type.
+- Note that we only support Tensorflow Estimator model, Keras model, and XGBoost model currently, and the
+`model_def.py` file is the model python definition only when the model is a Tensorflow Estimator model or Keras model. For XGBoost model, the content of `model_def.py` should be like: `model_type = xgboost.gbtree`.
+
+To publish a new model into the model zoo, you must:
+
+1. commit your model (code and README.md) to https://github.com/sql-machine-learning/models and merge the code.
+1. train and test your model on the dataset then save the pre-trained model together with `columns` file.
+1. name a directory and upload all files listed above into the directory on the server.
+
+## Using the Model Zoo in SQLFlow Statements
+
+In SQLFlow, you can specify the model under `sqlflow.org/modelzoo` in the `TRAIN` and `USING` clause.
+
+If the `TRAIN` clause accepts a model under `sqlflow.org/modelzoo`, SQLFlow will only use the model definition to start the train. If `USING` clause accepts a model under `sqlflow.org/modelzoo`, the model's
```

Comment:
How do we specify what engine to be used? Should this be part of the model name or is there a better way to explicitly specify this?

Reply:
You are right, engine type, as well as engine version, is needed as the model's metadata. I'll add a file `engine.meta` in the model directory.